src/palladio/SOPAssign.cpp

```diff
@@ -462,6 +465,27 @@ bool evaluateDefaultRuleAttributes(SOPAssign* node, const GU_Detail* detail, Sha
 	return true;
 }
 
+template<typename T>
+std::pair<UT_ValArray<T>, bool> getArrayFromMultiParm(const SOPAssign* const node, const PRM_Parm& parm, PRM_Type expectedType, std::function<T(const SOPAssign*, const PRM_Parm*)> eval) {
```

Comment:
consider replacing std::function with a template argument (std::function has runtime overhead and is more verbose)

```
template<typename T, typename E>
std::pair<UT_ValArray<T>, bool> getArrayFromMultiParm(const SOPAssign* const node, const PRM_Parm& parm, PRM_Type expectedType, E eval) {
```

Reply:
done